official/boosted_trees_upgraded/data_download.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Downloads the UCI HIGGS Dataset and prepares train data.
+
+The details on the dataset are in https://archive.ics.uci.edu/ml/datasets/HIGGS
+
+It takes a while as it needs to download 2.8 GB over the network, process, then
+store it into the specified location as a compressed numpy file.
+
+Usage:
+$ python data_download.py --data_dir=/tmp/higgs_data
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import gzip
+import os
+import tempfile
+
+# pylint: disable=g-bad-import-order
+import numpy as np
+import pandas as pd
+from six.moves import urllib
+from absl import app as absl_app
+from absl import flags,logging
+import tensorflow as tf
+
+from official.utils.flags import core as flags_core
+
+URL_ROOT = "https://archive.ics.uci.edu/ml/machine-learning-databases/00280"
+INPUT_FILE = "HIGGS.csv.gz"
+NPZ_FILE = "HIGGS.csv.gz.npz"  # numpy compressed file to contain "data" array.
+
+
+def _download_higgs_data_and_save_npz(data_dir):
+  """Download higgs data and store as a numpy compressed file."""
+  input_url = URL_ROOT + "/" + INPUT_FILE
+  np_filename = os.path.join(data_dir, NPZ_FILE)
+  if tf.io.gfile.exists(np_filename):
+    raise ValueError("data_dir already has the processed data file: {}".format(
+        np_filename))
+  if not tf.io.gfile.exists(data_dir):
+    tf.io.gfile.mkdir(data_dir)
+  # 2.8 GB to download.
+  try:
+    tf.compat.v1.logging.info("Data downloading...")
+    temp_filename, _ = urllib.request.urlretrieve(input_url)
+    # Reading and parsing 11 million csv lines takes 2~3 minutes.
+    tf.compat.v1.logging.info("Data processing... taking multiple minutes...")
```

Comment:
could use get_logger. Same for all logging related.

Reply:
is get_logger a utility?..I didn't find it. I used absl.logging. Maybe I forgot to replace it in this file. Can I replace it with absl.logging?